src/main/java/fi/metatavu/metamind/bot/BotController.java

```diff
@@ -110,35 +111,50 @@ public MetamindBot getBotInstance(Story story) {
     String storyJson = story.getStoryJson();
     
     Map<String, SlotModel> slotModels = new HashMap<>();
-    IntentModel intentModel = null;
+    List<IntentModel> intentModels = new ArrayList<IntentModel>();
+    List<MachineLearningConfig> machineLearningConfigs = storyConfig.getMachineLearning();
     
     if (storyConfig.getMachineLearning() != null) {
       loadSlotModels(storyConfig, slotModels);
       
-      String intentModelName = storyConfig.getMachineLearning().getIntentModel();
-      if (intentModelName != null) {
-        intentModel = modelsContoller.findIntentModelByName(String.format("%s.bin", intentModelName));
-        if (intentModel == null && logger.isWarnEnabled()) {
-          logger.warn(String.format("Failed to load intent model %s", intentModelName));
+      List<String> intentModelNames = new ArrayList<String>();
+      
+      for (MachineLearningConfig machineLearningConfig : machineLearningConfigs) {
+        intentModelNames.add(machineLearningConfig.getIntentModel());
+      }
+      
+      for (String intentModelName : intentModelNames) {
+        if (intentModelName != null) {
+          IntentModel intentModel = modelsContoller.findIntentModelByName(String.format("%s.bin", intentModelName));
+          
+          if (intentModel == null && logger.isWarnEnabled()) {
+            logger.warn(String.format("Failed to load intent model %s", intentModelName));
+          } else {
+            intentModels.add(intentModel);
+          }
         }
       }
     }
     
-    MetamindBotConfiguration botConfiguration = new MetamindBotConfiguration(storyConfig, functions, storyJson, intentModel, slotModels);
+    MetamindBotConfiguration botConfiguration = new MetamindBotConfiguration(storyConfig, functions, storyJson, intentModels, slotModels);
     
     return new MetamindBot(botConfiguration);
   }
 
   protected void loadSlotModels(StoryConfig storyConfig, Map<String, SlotModel> slotModels) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)